src/broadcast/mpris/mediaplayer2.cpp

```diff
@@ -0,0 +1,72 @@
+#include "broadcast/mpris/mediaplayer2.h"
+
+#include <QApplication>
+#include <QDebug>
+#include <QString>
+#include <QWidget>
+
+#include "mixxx.h"
+#include "moc_mediaplayer2.cpp"
+#include "sources/soundsourceproxy.h"
+
+MediaPlayer2::MediaPlayer2(MixxxMainWindow* pMixxx, QObject* parent)
+        : QDBusAbstractAdaptor(parent),
+          m_pMixxx(pMixxx) {
+}
+
+bool MediaPlayer2::canQuit() const {
+    return true;
+}
+
+bool MediaPlayer2::fullscreen() const {
+    return m_pMixxx->isFullScreen();
```

Comment:
How can we get this information without m_pMixxx? Or maybe remove it? Is it really needed?

Reply:
we could track the signals from the mainwindow whne switching from/to fullscreen

Reply:
I like that idea.

Reply:
Due to the connect call we get the coupling anyway. So there is not really a benefit using the complexity of signal and slots.